### PR TITLE
Update sklearn_regression submodule to point to mlr-regression-example

### PR DESCRIPTION
**Pull request by mingyu89**
Signed-off-by: Mingyu Li <mingyu.li@databricks.com>
As title.
Example repo that MLflow can use to run integration tests and demo for first time MLP users.

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [x] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by mingyu89** [comment]
@jinzhang21  Lots of failures about repo not found. Is it because it's private? https://github.com/mlflow/mlr-regression-example

**Comment by mingyu89** [comment]
Made it the repo public

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
 [submodule "examples/pipelines/sklearn_regression"]
 	path = examples/pipelines/sklearn_regression
 	url = https://github.com/mlflow/mlp-regression-template.git
```
Comment (jinzhang21) [inline .gitmodules:3]: Let's just point this url to the new example repo instead of adding a new one.

File: docs/source/pipelines.rst [hunk 2]
```diff
@@ -139,7 +139,7 @@ Pipeline Repositories
 
 Pipelines are structured as git repositories with a standardized directory hierarchy. The following
 example provides an overview of the pipeline repository structure.  It is adapted from the
-|MLflow Regression Pipeline repository|.
+|MLflow Recipe Regression Example repository|.
```
Comment (jinzhang21) [inline docs/source/pipelines.rst:142]: Can we make the doc change in a separate PR? We haven't fully figured out the name yet. I don't want to block this PR because of that.

**Review by jinzhang21 — APPROVED** [review]
Feel free to merge after addressing the comment. Thanks, @mingyu89 !

**Review by jinzhang21 — COMMENTED** [review]
Looks like unit tests are failing. Please take a look first.

**Comment by mingyu89** [comment]
Not able to reuse the same name. I'll modify it.

**Comment by mingyu89** [comment]
Github probably took it as a commit(HEAD) change instead of repo change.

File: .gitmodules [hunk 3]
```diff
@@ -1,3 +1,3 @@
-[submodule "examples/pipelines/sklearn_regression"]
-	path = examples/pipelines/sklearn_regression
-	url = https://github.com/mlflow/mlp-regression-template.git
+[submodule "examples/pipelines/sklearn_regression_example"]
+	path = examples/pipelines/sklearn_regression_example
```
Comment (jinzhang21) [inline .gitmodules:2]: We probably have to rename the directory later, but let's leave it as `pipelines` for now.

**Review by jinzhang21 — APPROVED** [review]
Feel free to merge if tests pass.

**Comment by mingyu89** [comment]
The PR of changing sklearn version in the example repo didn't work.
This is because the version while installing MLP is still 1.0.X and that's the cause of the error.
This is the command
```
pip install -e '.[pipelines]'
```
Screen Shot 2022-09-29 at 11 09 09 PM

I have to remove the feature_names_out from the example repo.
